zarr/core.py

```diff
@@ -1950,7 +1950,11 @@ def _process_for_setitem(self, ckey, chunk_selection, value, fields=None):
         return self._encode_chunk(chunk)
 
     def _chunk_key(self, chunk_coords):
-        return self._key_prefix + '.'.join(map(str, chunk_coords))
+        if hasattr(self._store, 'key_separator'):
```

Comment:
A heads up that #716 deprecates `key_separator` in favor `store._dimension_separator`. Looking at this usage, I imagine wrapping stores like `LRUCacheStore` and `ConsolidatedMetadataStore` are going to need to set `_dimension_separator as well.

Reply:
Okay, thanks for letting me know. Should we hold off on this PR until #716 is in and then revisit?

Reply:
Nah, let's get this out as a point release. Just a warning that this "API" will change. It's all kismet.

Reply:
The behavior of `_chunk_key` should not be changed here. Instead, the behavior of `FSStore._getitems_` should be changed to properly remap the keys. See https://github.com/zarr-developers/zarr-python/pull/709/files#diff-565e487a2f60258b6baa2e4db8ef175cc16b8a949651834bd43d0a9f21e07358R1054